src/drawer.multicanvas.js

```diff
@@ -457,25 +470,36 @@ export default class MultiCanvas extends Drawer {
      * @param {number?} end The x-offset of the end of the area that should be
      * rendered
      * @param {function} fn The render function to call, e.g. `drawWave`
+     * @param {number} drawIndex The index of the current channel after filtering.
      * @returns {void}
      */
-    prepareDraw(peaks, channelIndex, start, end, fn) {
+    prepareDraw(peaks, channelIndex, start, end, fn, drawIndex) {
         return util.frame(() => {
             // Split channels and call this function with the channelIndex set
             if (peaks[0] instanceof Array) {
                 const channels = peaks;
+
                 if (this.params.splitChannels) {
-                    this.setHeight(
-                        channels.length *
-                            this.params.height *
-                            this.params.pixelRatio
-                    );
-                    return channels.forEach((channelPeaks, i) =>
-                        this.prepareDraw(channelPeaks, i, start, end, fn)
-                    );
+                    const filteredChannels =  channels.filter((c, i) => !this.hideChannel(i));
+                    if (!this.params.splitChannelsOptions.overlay) {
+                        this.setHeight(
+                            Math.max(filteredChannels.length, 1) *
+                                this.params.height *
+                                this.params.pixelRatio
+                        );
+                    } 
+
+                    return channels.forEach((channelPeaks, i) => 
+                        this.prepareDraw(channelPeaks, i, start, end, fn, filteredChannels.indexOf(channelPeaks))
+                    );                    
                 }
                 peaks = channels[0];
             }
+
+            if (this.hideChannel(channelIndex)) {
```

Comment:
can you add a comment here explaining why it returns

Reply:
done. let me know if it's still unclear.